CMakeLists.txt

```diff
@@ -19,12 +21,14 @@ list(APPEND CMAKE_MODULE_PATH
 
 add_subdirectory(extern)
 
-option(BUILD_TESTING "Run unit tests only." OFF)
```

Comment:
This line was here for a reason. Without it, `include(CTest)` set `BUILD_TESTING` to `ON` and breaks `mbed-tools compile` of applications

Reply:
Good catch. This is because our examples define the application project after adding the mbed-os subdirectory. So the mbed-os project is considered the "current project" when this file is parsed. If I make the following change to e.g mbed-os-example-blinky's CMakeLists.txt it compiles fine here:

```

 include(${MBED_PATH}/tools/cmake/app.cmake)

+project(${APP_TARGET})
+
 add_subdirectory(${MBED_PATH})

 add_executable(${APP_TARGET})

 mbed_configure_app_target(${APP_TARGET})

-project(${APP_TARGET})
```

I think the correct thing would be to fix the examples so they define their project before adding the mbed-os subdirectory.

Reply:
That makes sense. Shall we fix all examples as a priority, then get this PR merged? Or merge this with the old workaround, fix examples at some point, then come back to remove the workaround?